Apps/Medication/src/Delegates/SalesforeceDelegate.cs

```diff
@@ -0,0 +1,160 @@
+//-------------------------------------------------------------------------
+// Copyright © 2019 Province of British Columbia
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//-------------------------------------------------------------------------
+namespace HealthGateway.Medication.Delegates
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Diagnostics;
+    using System.Net;
+    using System.Net.Http;
+    using System.Net.Http.Headers;
+    using System.Net.Mime;
+    using System.Text.Json;
+    using System.Threading.Tasks;
+    using HealthGateway.Common.AccessManagement.Authentication;
+    using HealthGateway.Common.Delegates;
+    using HealthGateway.Common.ErrorHandling;
+    using HealthGateway.Common.Models;
+    using HealthGateway.Common.Services;
+    using HealthGateway.Medication.Models;
+    using Microsoft.Extensions.Configuration;
+    using Microsoft.Extensions.Logging;
+
+    /// <summary>
+    /// Salesforce Implementation that retrieves Medication Requests.
+    /// </summary>
+    public class SalesforeceDelegate : IMedicationRequestDelegate
+    {
+        private const string AuthConfigSectionName = "ClientAuthentication";
+        private const string SalesForceConfigSectionKey = "Salesforce";
```

Comment:
Suggest renaming constant to `SalesforceConfigSectionKey`.

Reply:
sounds good. Fixed